### PR TITLE
[MRG] Code RAG for Chatbot

### LINKED ISSUE #259: Enhance Local Code Generation with RAG Module
- Complete a PoC with LanceDB by implementing a memory.py file.
- Establish a workdir for `mle chat` and `mle start` (baseline mode) where code can be scanned to generate embeddings for RAG. Start by evaluating embedding models, with a recommendation to begin with OpenAI.

### PR DESCRIPTION
**Pull request by huangyz0918**
Closes #259

- using the `tree-sitter` to chunk the code into snippets
- applied the memory into the `mle-chat`

### CODE REVIEW

**Comment by huangyz0918** [comment]
Try `mle chat --build_mem` on an mle-agent project

**Review by HuaizhengZhang — APPROVED** [review]
LGTM.

Some minor naming or doc issues

File: mle/cli.py [hunk 1]
```diff
@@ -187,14 +191,42 @@ def kaggle(
...
+        chunker = CodeChunker(os.path.join(working_dir, '.mle', 'cache'), 'py')
+        with Progress(
+                SpinnerColumn(),
+                TextColumn("[progress.description]{task.description}"),
+                BarColumn(),
+                TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
+                TimeElapsedColumn(),
+                console=console,
+        ) as progress:
+            process_task = progress.add_task("Processing files...", total=len(source_files))
```
[the remark this answers is not in the document]
Comment (huangyz0918) [reply mle/cli.py:216]: I changed it to "add <file_name>.py into memory..."

File: mle/agents/chat.py [hunk 2]
```diff
@@ -56,7 +57,9 @@ def __init__(self, model, working_dir='.', console=None):
             schema_search_papers_with_code,
             schema_web_search,
             schema_execute_command,
-            schema_preview_csv_data
+            schema_preview_csv_data,
```
[the remark this answers is not in the document]
Comment (huangyz0918) [reply mle/agents/chat.py:60]: Interesting idea, we can design a function store later

**Pull request merged**